accepted/future-releases/enhanced-enums/feature-specification.md

```diff
@@ -152,20 +155,18 @@ It’s currently a compile-time error for a class to implement, extend or mix-in
 
-- It's a compile-time error if a *non-abstract* class has `Enum` as a superinterface (directly or transitively) unless it is the corresponding class of an `enum` declaration.
+- It's a compile-time error if a *non-abstract* class has `Enum` as a superinterface (directly or transitively) unless it is the corresponding class of an `enum` declaration. _(Abstract interfaces and mixins implementing `Enum` are allowed, but only so that they can be used by `enum` declarations, they can never be used to create an instance which implements `Enum`, but which is not an enum value.)_
 - It is a compile-time error if a class implements, extends or mixes-in the class or interface introduced by an `enum` declaration. _(An enum class can’t be used as a mixin since it is not a `mixin` declaration and the class has a superclass other than `Object`, but we include “mixes-in” for completeness.)_
 - It's a **compile-time error** if a `class` or `mixin` declaration has `Enum` as a superinterface and the interface of the declarations contains an instance member with the name `values`, whether declared or inherited. _If any concrete class implements this interface, it will be an `enum` declaration class, and then the `values` member would conflict with the static `values` constant getter that is automatically added to `enum` declaration classes. Such an instance `values` declaration is either useless or wrong, so we disallow it entirely._
-- It's a compile-time error if a `class`, `mixin` or `enum` declaration has `Enum` as a superinterface, and it declares a non-abstract instance member named `index`. _That member would override the `index` getter inherited from `Enum`, and we currently do not allow that._
+- It's a compile-time error if a `class`, `mixin` or `enum` declaration has `Enum` as a superinterface (trivially true for `enum` declarations), and that declaration contains non-abstract instance member declaration named `index`, `hashCode` or `==` (an `operator ==` declaration). _That `index` member would override the `index` getter inherited from `Enum`, and we currently do not allow that. The `hashCode` and `operator==` declarations would prevent the enum class from having "primitive equality", and we want to ensure that enums can be used in switches._
 
-Those restrictions allows abstract classes (interfaces) which implements `Enum` in order to have the `int index;` getter member available, and it allows `mixin` declarations to use `Enum` as an `on` type because `mixin` declarations cannot be instantiated directly.
+Those restrictions allow abstract classes (interfaces) which *implements* `Enum` in order to have the `int index;` getter member available, and also allow `mixin` declarations to use `Enum` as an `on` type because `mixin` declarations cannot be instantiated directly.
```

Comment:
implements -> implement

Reply:
The language specification uses `\IMPLEMENTS` as a reference to the actual syntax and 'implements' as a separate concept (such that `List<int>` implements `Iterable<int>` and `Object`, but not `Iterable<num>`). In this context I think we could use 'abstract classes that implement `Enum`' and remain consistent with the way this word is used in the spec.

We could include the word 'interface', but I think that's more opinion than information. ;-)